qiskit/providers/aer/primitives/estimator.py

```diff
@@ -0,0 +1,151 @@
+from qiskit.primitives.utils import init_circuit, init_observable
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .. import AerSimulator
+from .base_estimator import BaseEstimator  # TODO: fix import path after Terra 0.21.
+
+
+class Estimator(BaseEstimator):
+    """
+    Aer implmentation of Estimator.
+
+    :Run Options:
+        - **shots** (None or int) --
+          The number of shots. If None, it calculates the exact expectation
+          values. Otherwise, it samples from normal distributions with standard errors as standard
+          deviations using normal distribution approximation.
+
+        - **seed_primitive** (np.random.Generator or int) --
+          Set a fixed seed or generator for rng. If shots is None, this option is ignored.
+    """
+
+    def __init__(
+        self,
+        circuits: QuantumCircuit | Iterable[QuantumCircuit],
+        observables: BaseOperator | PauliSumOp | Iterable[BaseOperator | PauliSumOp],
+        parameters: Iterable[Iterable[Parameter]] | None = None,
+        backend_options: dict | None = None,
+        transpile_options: dict | None = None,
+    ):
+        """
+        Args:
+            circuits: Quantum circuits that represent quantum states.
+            observables: Observables.
+            parameters: Parameters of quantum circuits, specifying the order in which values
+                will be bound. Defaults to ``[circ.parameters for circ in circuits]``
+                The indexing is such that ``parameters[i, j]`` is the j-th formal parameter of
+                ``circuits[i]``.
+            backend_options: Options passed to AerSimulator.
+            transpile_options: Options passed to transpile.
+        """
+        if isinstance(circuits, QuantumCircuit):
+            circuits = (circuits,)
+        circuits = tuple(init_circuit(circuit) for circuit in circuits)
+
+        if isinstance(observables, (PauliSumOp, BaseOperator)):
+            observables = (observables,)
+        observables = tuple(init_observable(observable) for observable in observables)
+
+        super().__init__(
+            circuits=circuits,
+            observables=observables,
+            parameters=parameters,
+        )
+        self._is_closed = False
+        self._backend = AerSimulator()
+        backend_options = {} if backend_options is None else backend_options
+        self._backend.set_options(**backend_options)
+        self._transpile_options = {} if transpile_options is None else transpile_options
+
+    def _call(
+        self,
+        circuits: Sequence[int],
+        observables: Sequence[int],
+        parameter_values: Sequence[Sequence[float]],
+        **run_options,
+    ) -> EstimatorResult:
+        if self._is_closed:
+            raise QiskitError("The primitive has been closed.")
+
+        shots = run_options.pop("shots", None)
+        seed_primitive = run_options.pop("seed_primitive", None)
```

Comment:
If `Estimator` takes into accounts of noise, both `seed_primitive` and `seed_simulator` must be set for reproducible results.  Can we automatically set one if another is specified?

Reply:
I intend to make it so that seed is selected in the order of 1. seed_primitive, 2. seed_simulator, and 3. default. I would like to write about this order of precedence in the documentation.

Reply:
Added in 01955c3